packages/iov-jsonrpc/src/parse.ts

```diff
@@ -0,0 +1,90 @@
+import { JsonRpcErrorResponse, JsonRpcRequest, JsonRpcResponse } from "./types";
+
+export function parseJsonRpcId(data: any): number | null {
+  const id = data.id;
+  if (typeof id !== "number") {
+    return null;
+  }
+  return id;
+}
+
+export function parseJsonRpcRequest(data: any): JsonRpcRequest {
+  if (typeof data !== "object") {
+    throw new Error("Data must be an object");
+  }
+
+  if (data.jsonrpc !== "2.0") {
+    throw new Error(`Got unexpected jsonrpc version: ${data.jsonrpc}`);
+  }
+
+  const id = parseJsonRpcId(data);
+  if (id === null) {
+    throw new Error("Invalid id field");
+  }
+
+  const method = data.method;
+  if (typeof method !== "string") {
+    throw new Error("Invalid method field");
+  }
+
+  const params = data.params;
+  if (!Array.isArray(params)) {
+    throw new Error("Invalid params field");
+  }
+
+  return {
+    jsonrpc: "2.0",
+    id: id,
+    method: method,
+    params: params,
+  };
+}
+
+export function parseJsonRpcError(data: any): JsonRpcErrorResponse | undefined {
```

Comment:
`data: unknown`?

Reply:
In case of objects I have the following ptoblem: using runtime checks I narrow the type from `unknown` to `object`. But than I cannot use `data: object` because of

```
src/parse.ts:48:12 - error TS2339: Property 'jsonrpc' does not exist on type 'object'.

48   if (data.jsonrpc !== "2.0") {
              ~~~~~~~
```

I don't know how to specify the type "any object"